test/primer/styleable_test.rb

```diff
@@ -0,0 +1,279 @@
+    assert_equal "ml-n4", EmptyComponent.new.ml(-4).primer_classes
+    assert_equal "mb-n4", EmptyComponent.new.mb(-4).primer_classes
+    assert_equal "mr-n4", EmptyComponent.new.mr(-4).primer_classes
+
+    assert_raises ArgumentError do
+      EmptyComponent.new.m(-1)
+    end
+
+    # positive over 6
+    assert_raises ArgumentError do
+      EmptyComponent.new.mr(7)
+    end
+
+    # negative under -6
+    assert_raises ArgumentError do
+      EmptyComponent.new.mr(-7)
+    end
+  end
+
+  def test_multiple_params
+    assert_equal "m-4 py-2", EmptyComponent.new.m(4).py(2).primer_classes
+  end
+
+  def test_font_size
+    assert_equal "f00", EmptyComponent.new.f("00").primer_classes
+    assert_equal "f0", EmptyComponent.new.f(0).primer_classes
+    assert_equal "f1", EmptyComponent.new.f(1).primer_classes
+    assert_equal "f2", EmptyComponent.new.f(2).primer_classes
+    assert_equal "f3", EmptyComponent.new.f(3).primer_classes
+    assert_equal "f4", EmptyComponent.new.f(4).primer_classes
+    assert_equal "f5", EmptyComponent.new.f(5).primer_classes
+    assert_equal "f6", EmptyComponent.new.f(6).primer_classes
+  end
+
+  def test_p
+    assert_equal "p-4", EmptyComponent.new.p(4).primer_classes
+    assert_equal "px-4", EmptyComponent.new.px(4).primer_classes
+    assert_equal "py-4", EmptyComponent.new.py(4).primer_classes
+    assert_equal "pt-4", EmptyComponent.new.pt(4).primer_classes
+    assert_equal "pl-4", EmptyComponent.new.pl(4).primer_classes
+    assert_equal "pb-4", EmptyComponent.new.pb(4).primer_classes
+    assert_equal "pr-4", EmptyComponent.new.pr(4).primer_classes
+
+    assert_raises ArgumentError do
+      EmptyComponent.new.p(-1)
+    end
+
+    assert_raises ArgumentError do
+      EmptyComponent.new.p(7)
+    end
+  end
+
+  def test_position
+    # TODO maybe make these methods instead of accepting an argument?
+    # e.g. `EmptyComponent.new.relative`, `EmptyComponent.new.absolute`
+    assert_equal "position-relative", EmptyComponent.new.position(:relative).primer_classes
+    assert_equal "position-absolute", EmptyComponent.new.position(:absolute).primer_classes
+    assert_equal "position-fixed", EmptyComponent.new.position(:fixed).primer_classes
+
+    assert_equal "top-0", EmptyComponent.new.top0.primer_classes
+    assert_equal "bottom-0", EmptyComponent.new.bottom0.primer_classes
+    assert_equal "left-0", EmptyComponent.new.left0.primer_classes
+    assert_equal "right-0", EmptyComponent.new.right0.primer_classes
+  end
+
+  def test_display
+    assert_equal "d-block", EmptyComponent.new.block.primer_classes
+    assert_equal "d-none", EmptyComponent.new.none.primer_classes
+    assert_equal "d-inline", EmptyComponent.new.inline.primer_classes
+    assert_equal "d-inline-block", EmptyComponent.new.inline_block.primer_classes
+    assert_equal "d-table", EmptyComponent.new.table.primer_classes
+    assert_equal "d-table-cell", EmptyComponent.new.table_cell.primer_classes
```

Comment:
Should probably prefix these methods with `d_`?

Reply:
Or, could we do `d(:table_cell)`?

Reply:
> Or, could we do `d(:table_cell)`?

We totally could. I went with methods for this first attempt at an implementation instead of arguments since it will immediately raise an error instead of requiring us to validate the arguments. Thoughts?